flexmeasures/data/models/planning/tests/test_solver.py

```diff
@@ -1859,3 +1859,86 @@ def test_battery_storage_efficiency_sensor(
 
     scheduler_info = scheduler._prepare()
     assert all(scheduler_info[5][0]["efficiency"] == expected_efficiency)
+
+
+@pytest.mark.parametrize(
+    "sensor_name, expected_start, expected_end",
+    [
+        (
+            "soc-targets (1h)",
+            "14:00:00",
+            "14:45:00",
+        ),  # A value defined in a larger resolution is downsampled to match the power sensor resolution.
```

Comment:
Two things:
- From 1 hour to 15 minutes is *upsampling* (increases data density). The terms *frequency* and *resolution* express a `1/time` relationship, but, somewhat confusingly, we (and Pandas) express them in units of `time`. This makes the concept of a 'larger resolution' ambiguous (just consider the term 'high-res'). Nevertheless, this is clearly a case of upsampling.
- I don't understand what the target value is supposed to express yet. This sensor has a value for an event lasting one hour, starting from 14.00 and ending at 15.00. I would expect then that, for a 15-minute resolution power sensor, this would translate into the constraint that the SoC value needs to be reached at 14.00 and should stay there until 15.00, rather than until 14.45.

Reply:
> From 1 hour to 15 minutes is upsampling (increases data density). The terms frequency and resolution express a 1/time relationship, but, somewhat confusingly, we (and Pandas) express them in units of time. This makes the concept of a 'larger resolution' ambiguous (just consider the term 'high-res'). Nevertheless, this is clearly a case of upsampling.

Thanks for clarifying the concept for me.

> I don't understand what the target value is supposed to express yet. This sensor has a value for an event lasting one hour, starting from 14.00 and ending at 15.00. I would expect then that, for a 15-minute resolution power sensor, this would translate into the constraint that the SoC value needs to be reached at 14.00 and should stay there until 15.00, rather than until 14.45.

Currently, the resampling logic is handled by the `search_beliefs` method. I think we should keep the resampling behavior the same for all the sensors and describing these nuances in the documentation. In this case, they could have created three 1h events to cover for the previous and next hours. It's true that it's not the same but after all, they should have used perhaps a 15min resolution to have more accuracy of an instantaneous. 

Said that, I'm open to alternatives :D

Reply:
I believe this is resolved now?